test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,388 @@
+            {
+                channelOperationResult2 = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent2.get());
+        });
+
+
+    auto channelOperation3 = PushNotificationManager::CreateChannelAsync(remoteId3);
+
+    channelOperation3.Completed(
+        [&channelEvent3, &channelOperationResult3](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                auto channelUri = result.Channel().Uri();
+                auto channelExpiry = result.Channel().ExpirationTime();
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                channelOperationResult3 = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent3.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent2.get(), 960000 /* milliseconds */))
+    {
+        channelOperation2.Cancel();
+        channelOperationResult2 = ERROR_TIMEOUT;
+    }
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent3.get(), 960000 /* milliseconds */))
+    {
+        channelOperation3.Cancel();
+        channelOperationResult3 = ERROR_TIMEOUT;
+    }
+
+    return ((channelOperationResult2 == WPN_E_OUTSTANDING_CHANNEL_REQUEST) && (channelOperationResult3 == WPN_E_OUTSTANDING_CHANNEL_REQUEST));
+}
+
+bool RegisterActivator()
+{
+    try
+    {
+        PushNotificationActivationInfo info(
+            PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
+            winrt::guid("00000000-0000-0000-0000-000000000001"));
+
+        auto token = PushNotificationManager::RegisterActivator(info);
+        if (token.Cookie() == 0 || token.TaskRegistration() == nullptr)
+        {
+            return false;
+        }
+    }
+    catch (...)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool UnregisterActivator()
+{
+    try
+    {
+        PushNotificationActivationInfo info(
+            PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
+            winrt::guid("00000000-0000-0000-0000-000000000001"));
+
+        auto token = PushNotificationManager::RegisterActivator(info);
+        if (token.Cookie() == 0 || token.TaskRegistration() == nullptr)
+        {
+            return false;
+        }
+
+        PushNotificationManager::UnregisterActivator(token, PushNotificationRegistrationKind::ComActivator);
```

Comment:
If we are registering a task, we should unregister it. Right now this is only cleaning the COM registration.